packages/web/hooks/one-click-trading/use-one-click-trading-params.ts

```diff
@@ -47,74 +48,47 @@ export function getParametersFromOneClickTradingInfo({
+    useState<OneClickTradingTransactionParams>();
+  const apiUtils = api.useUtils();
+
   const {
     data: defaultTransaction1CTParams,
     isLoading,
     isError,
-  } = api.local.oneClickTrading.getParameters.useQuery();
-
-  const [transaction1CTParams, setTransaction1CTParams] = useState<
-    OneClickTradingTransactionParams | undefined
-  >(
-    oneClickTradingInfo
-      ? getParametersFromOneClickTradingInfo({
-          oneClickTradingInfo,
-          defaultIsOneClickEnabled,
-        })
-      : undefined
-  );
-  const [initialTransaction1CTParams, setInitialTransaction1CTParams] =
-    useState<OneClickTradingTransactionParams | undefined>();
-
-  useEffect(() => {
-    const paramsToSet = oneClickTradingInfo
-      ? getParametersFromOneClickTradingInfo({
-          oneClickTradingInfo,
-          defaultIsOneClickEnabled,
-        })
-      : defaultTransaction1CTParams;
+    refetch: reset,
+  } = useQuery({
+    queryKey: [scopeKey, oneClickTradingInfo?.sessionKey],
+    queryFn: async () => {
+      const data = await apiUtils.local.oneClickTrading.getParameters.fetch();
+      if (oneClickTradingInfo) {
+        return {
+          ...getParametersFromOneClickTradingInfo({
+            oneClickTradingInfo,
+            defaultIsOneClickEnabled,
+          }),
+          spendLimitTokenDecimals: data.spendLimitTokenDecimals,
+        };
+      }
 
-    if (!paramsToSet || transaction1CTParams) return;
-
-    const nextTransaction1CTParams = {
-      isOneClickEnabled: defaultIsOneClickEnabled,
-      ...paramsToSet,
-    };
-    setTransaction1CTParams(nextTransaction1CTParams);
-    setInitialTransaction1CTParams(nextTransaction1CTParams);
-  }, [
-    defaultIsOneClickEnabled,
-    defaultTransaction1CTParams,
-    oneClickTradingInfo,
-    transaction1CTParams,
-  ]);
-
-  const reset = useCallback(() => {
-    const paramsToSet = oneClickTradingInfo
-      ? getParametersFromOneClickTradingInfo({
-          oneClickTradingInfo,
-          defaultIsOneClickEnabled,
-        })
-      : defaultTransaction1CTParams;
-    if (!paramsToSet && !initialTransaction1CTParams) return;
-
-    const nextTransaction1CTParams = paramsToSet
-      ? {
-          isOneClickEnabled: defaultIsOneClickEnabled,
-          ...paramsToSet,
-        }
-      : initialTransaction1CTParams;
-    setTransaction1CTParams(nextTransaction1CTParams);
-  }, [
-    defaultIsOneClickEnabled,
-    defaultTransaction1CTParams,
-    initialTransaction1CTParams,
-    oneClickTradingInfo,
-  ]);
+      return {
+        ...data,
+        spendLimitTokenDecimals: data.spendLimitTokenDecimals,
+        isOneClickEnabled: defaultIsOneClickEnabled,
+      };
+    },
+    cacheTime: 1000 * 60 * 10,
+    refetchOnWindowFocus: false, // Prevents refetching after losing the focus when accepting the transaction
+    onSuccess: ({ spendLimitTokenDecimals, ...rest }) => {
+      setTransaction1CTParams(rest);
+    },
+  });
```

Comment:
We want to retain the previous parameters to track changes, so we avoided overriding the state. Also, these modifications have introduced a bug. Please review the videos for more details.

Before

https://github.com/user-attachments/assets/aeaddbd9-db4a-4dfa-a61f-2bdd360346ee


After

https://github.com/user-attachments/assets/4e13b585-2e69-4c7f-87e2-ac14a1fde6c8

Initiating a transaction or any state change triggers a parameter update, causing the pending changes to be lost. This is unintended, as it prevents the discard dialog from appearing. Additionally, we want to avoid pre-emptive optimizations, as they tend to increase the risk of bugs and complicate the implementation for scenarios that are unlikely to occur.